src/objective/rank_obj.cc

```diff
@@ -170,14 +173,32 @@ class LambdaRankObj : public ObjFunction {
   virtual void GetLambdaWeight(const std::vector<ListEntry> &sorted_list,
                                std::vector<LambdaPair> *io_pairs) = 0;
 
- private:
+  void Save(Json* p_out) const override {
+    auto& out = *p_out;
+    out["name"] = String("LambdaRankObj");
+    out["lambda_rank_param"] = Object();
+    for (auto const& kv : param_.__DICT__()) {
+      out["lambda_rank_param"][kv.first] = kv.second;
+    }
+  }
+
+ protected:
   LambdaRankParam param_;
 };
 
 class PairwiseRankObj: public LambdaRankObj{
  protected:
   void GetLambdaWeight(const std::vector<ListEntry> &sorted_list,
                        std::vector<LambdaPair> *io_pairs) override {}
+
+  void Save(Json* p_out) const override {
+    auto&  out = *p_out;
+    out["name"] = String("rank:pairwise");
+    out["lambda_rank_param"] = toJson(LambdaRankObj::param_);
```

Comment:
I noticed that here the key name is `lambda_rank_param` but above `SoftmaxMultiClassParam` is used. Do we have a naming convention we want to follow for the produced json?
I would personally prefer `lower_case` as opposite to `CamelCase` but I would rather have the single convention then none :)

Reply:
@trams Will unify it.

Reply:
@trams  I modified to snake case in  #4945 .  I'm also tracking your comment in this PR.  #4945 and #4936 will lighten this PR a lot, but the critical part is still in this PR (saving the model itself).  Also, all your comments are either not related to those PRs or addressed.  Thanks for taking your time with this.  Let me know your thoughts.